quickwit-search/src/retry/search_stream.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use quickwit_proto::{LeafSearchStreamRequest, LeafSearchStreamResult};
+use tokio::sync::mpsc::error::SendError;
+
+use super::RetryPolicy;
+use crate::SearchError;
+
+/// Retry policy for consuming the result stream of a LeafSearchStreamRequest.
+/// A retry is only made if there are some missing splits.
+/// As errors only come from a closed receiver, we ignore them.
+pub struct LeafSearchStreamRetryPolicy {}
+
+pub type SuccessFullSplitIds = Vec<String>;
```

Comment:
Nitpick: No need to fix, but typedef in rust are quite a bummer. They are like C++ typedef. It is just a type alias.

If you want static typing over it, you can do

```pub struct SuccessfullSplitIds(pub Vec<String>);```

Reply:
SuccessfulSplitIds